src/ajax/ring.cljc

```diff
@@ -0,0 +1,31 @@
+(ns ajax.ring
+  (:require [ajax.interceptors :refer [map->ResponseFormat]]
+            [ajax.protocols :refer [-status -body]]
+            #?@ (:cljs [[cljs.reader :as edn]]
+                 :clj [[clojure.edn :as edn]
+                       [clojure.java.io :refer [reader]]])))
+
+(defn make-ring-read [body-read]
+  (fn ring-read [xhrio]
+    {:status (-status xhrio)
+     :headers #? (:clj (reduce (fn [headers header]
```

Comment:
Thinking the "get all headers" behaviour might be better put into the AjaxResponse protocol. This is more efficient, but having a "get a list of tuples" method would be more generally useful to people.